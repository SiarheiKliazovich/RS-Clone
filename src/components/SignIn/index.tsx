import { FunctionComponent } from "react";
import FormSignIn from "../FormSignIn";

import "./signin.scss";

const SignIn: FunctionComponent = () => (
  <div className="sign__wrapper">
    <div className="sign__image">
      <div className="image-wrapper"></div>
    </div>
    <div className="sign__content-wrapper">
      <FormSignIn />
    </div>
  </div>
);

export default SignIn;
