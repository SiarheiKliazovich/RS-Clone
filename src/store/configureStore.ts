/* eslint-disable @typescript-eslint/ban-ts-comment */
import { applyMiddleware, createStore } from "redux";
import thunkMiddleware from "redux-thunk";

import loggerMiddleware from "./logger";
import rootReducer from "../reducers";

const configureStore = () => {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(rootReducer, middlewareEnhancer);

  // @ts-ignore:next-line
  if (process.env.NODE_ENV !== "production" && module.hot) {
    // @ts-ignore:next-line
    module.hot.accept("./../reducers", () => store.replaceReducer(rootReducer));
  }

  return store;
};

export default configureStore;
