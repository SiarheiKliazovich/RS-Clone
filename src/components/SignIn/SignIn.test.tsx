import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import SignIn from ".";

const mockStore = configureStore([]);

describe("SignIn", () => {
  // eslint-disable-next-line
  let store: any;

  beforeEach(() => {
    store = mockStore({
      appInterface: {
        accentColor: {
          static: "#000",
          hover: "#fff",
        },
      },
    });
  });

  test("renders correctly", () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <SignIn />
        </Router>
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
