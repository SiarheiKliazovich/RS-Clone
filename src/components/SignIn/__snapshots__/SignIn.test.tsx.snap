// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn renders correctly 1`] = `
<div
  class="sign__wrapper"
>
  <div
    class="sign__image"
  >
    <div
      class="image-wrapper"
    />
  </div>
  <div
    class="sign__content-wrapper"
  >
    <div
      class="sign__content"
    >
      <h2
        class="sign__title"
      >
        sign.signIn
      </h2>
      <div
        class="sign__text"
      >
        sign.text-1
      </div>
      <form
        action="#"
        class="form"
      >
        <input
          class="field"
          id="email"
          name="email"
          placeholder="Email"
          type="email"
          value=""
        />
        <input
          class="field"
          id="password"
          name="password"
          placeholder="Password"
          type="password"
          value=""
        />
        <button
          class="sign__button"
          style="background-color: rgb(0, 0, 0);"
          type="submit"
        >
          sign.signIn
        </button>
      </form>
      <div
        class="sign__text-link"
      >
        sign.create
         
        <a
          class="text-link"
          href="/sign-up"
          style="color: rgb(0, 0, 0);"
        >
          sign.signUp
        </a>
      </div>
    </div>
  </div>
</div>
`;
